spec.bs

```diff
@@ -501,14 +506,59 @@ are:
     1. Let |permission| be the result of [=checking interest group permissions=] with
       |owner|, |frameOrigin|, and "`leave`".
     1. If |permission| is false, then [=queue a task=] to [=reject=] |p| with a
-       "{{NotAllowedError}}" {{DOMException}} and do not run the remaining steps.
+       "{{NotAllowedError}}" {{DOMException}} and abort these steps.
     1. [=Queue a task=] to [=resolve=] |p| with `undefined`.
     1. [=list/Remove=] [=interest groups=] from the user agent's [=interest group set=] whose
       [=interest group/owner=] is |owner| and [=interest group/name=] is |name|.
 1. Return |p|.
 
 </div>
 
+<h3 id="clearoriginjoinedAdInterestGroups">clearOriginJoinedAdInterestGroups()</h3>
+{{Window/navigator}}.{{Navigator/clearOriginJoinedAdInterestGroups()}} removes a user from
+[=interest groups=] whose [=interest group/joining origin=] is the current top level page's
+origin.
+
+
+<xmp class="idl">
+[SecureContext]
+partial interface Navigator {
+  Promise<undefined> clearOriginJoinedAdInterestGroups(
+      USVString owner, optional sequence<USVString> interestGroupsToKeep = []);

```

Comment:
I noticed that these are both USVStrings instead of DOMStrings both here and in the implementation. Is `owner` or any of the groups-to-keep URLs (and thus _that's_ why they need to be USVStrings specifically).

Reply:
owner is an origin, not a URL.  interestGroupsToKeep is a list of interest group names.  The values correspond exactly to "USVString owner" and "USVString name" in the struct GenerateBidInterestGroup, so the types should be the same.